src/main/java/org/jabref/model/entry/BibtexEntryTypes.java

```diff
@@ -287,4 +287,12 @@ private BibtexEntryTypes() {
     public static Optional<EntryType> getType(String name) {
         return ALL.stream().filter(e -> e.getName().equalsIgnoreCase(name)).findFirst();
     }
+
+    public static EntryType getTypeOrDefault(String name) {
+        Optional<EntryType> optType = getType(name);
+        if (optType.isPresent()) {
+            return optType.get();
+        }
+        return new CustomEntryType(name,"required","optional");
```

Comment:
I can do that for sure. There is, however a slightly semantic difference. As is you know have the possibility to differentiate between an entry type that is present in the enumeration from one that is defaulted and returned custom. If we merge this logic in the `getType` method we loose this capability because we will always return something. Or a value from the enumerator or the defaulted one. I'm not the best person to judge if this is a good decision or not because I'm a newbie in the code however if that capability is not needed nor desired I agree that is a good way to cleanup code. What do you think?

Reply:
Hi! If I got it right, @linusdietz suggested to replace your code with the one-liner in the `getTypeOrDefault()` method.

Best,
Matthias